cvat-ui/src/components/selectors/organization-selector.tsx

```diff
@@ -0,0 +1,108 @@
+// Copyright (C) CVAT.ai Corporation
+//
+// SPDX-License-Identifier: MIT
+
+import './styles.scss';
+import _ from 'lodash';
+import React, { useEffect, useState } from 'react';
+import AutoComplete from 'antd/lib/auto-complete';
+
+import { Organization } from 'cvat-core-wrapper';
+import { getOrganizationsAsync } from 'actions/organization-actions';
+import { useSelector, useDispatch } from 'react-redux';
+import { CombinedState } from 'reducers';
+
+function OrganizationSelector(props: {
+    setNewOrganization: (org: Organization | null) => void;
+    defaultValue?: string
+}): JSX.Element {
+    const {
+        setNewOrganization,
+        defaultValue,
+    } = props;
+
+    const defaultPage = 1;
+    const [page, setPage] = useState(defaultPage);
+    const [searchPhrase, setSearchPhrase] = useState(defaultValue || '');
+    const [searchResults, setSearchResults] = useState<Organization[]>([]);
+
+    const organizations = useSelector((state: CombinedState) => state.organizations.currentArray);
+    const fetching = useSelector((state: CombinedState) => state.organizations.currentArrayFetching);
+    const hasMore = useSelector((state: CombinedState) => Boolean(state.organizations.nextPageUrl));
+    const showSandboxOption = useSelector((state: CombinedState) => Boolean(state.organizations.current));
+
+    const dispatch = useDispatch();
+
+    useEffect(() => {
+        if (organizations.length) {
+            setSearchResults((prev) => (page === defaultPage ? [...organizations] : [...prev, ...organizations]));
+        }
+    }, [organizations]);
+
+    useEffect(() => {
+        setSearchResults([]);
+        const filterQuery = {
+            page: defaultPage,
+            search: searchPhrase,
+        };
+        dispatch(getOrganizationsAsync(filterQuery, true));
+        setPage(defaultPage);
+    }, [searchPhrase]);
+
+    useEffect(() => {
+        if (page === defaultPage) return;
+
+        const filterQuery = {
+            page,
+            ...((searchPhrase) ? { search: searchPhrase } : {}),
+        };
+        dispatch(getOrganizationsAsync(filterQuery, true));
+    }, [page]);
+
+    const handlePopupScroll = (e: React.UIEvent<HTMLDivElement>) => {
+        const target = e.currentTarget;
+        if (
+            ((target.scrollTop + target.offsetHeight) >= (target.scrollHeight - 10)) &&
```

Comment:
Instead of relying on scroll events, I’d suggest using an Intersection Observer for the last element in the list. We can leverage `useInView` from `react-intersection-observer`.

One approach is to place a `ref` on the last label in the options, or alternatively, add a static invisible option at the end of the list to act as the observer target.